### ISSUE TITLE
Encoder or decoder structure?

### ISSUE THREAD

**Issue by mgcyung** [bowang-lab/scGPT#27]
Which transformer structure does the scGPT use, encoder (like Bert) or decoder (like GPT)?

**Comment by subercui** [comment]
Hi, we designed our own attention mask strategy for the pre-training stage, so it is not exactly an encoder or decoder. Please find the description in methods 4.3 in our new preprint. In the fine-tuning stage, for the tasks we listed so far, we didn't use the customized attention mask and it then behaves more like encoder.

**Comment by mgcyung** [comment]
Why the customized attention mask is not used in the fine-tuning stage? Is it used in the inference stage?
